### DISCUSSION TITLE
Route created but how to implement this code this is right way? Can you please help me.

### DISCUSSION THREAD

**Discussion by alok2911** [imdhemy/laravel-in-app-purchases#258] [Q&A]
Route is 
Route::post('/liap/notifications', 'UserController@appStoreNotification')->name('liap.serverNotifications');

public function appStoreNotification(Request $request)
    {
        try {

            $receiptResponse = Subscription::appStore()->receiptData($receipt)->verify();

           

            // Get the receipt status
            $receiptStatus = $receiptResponse->getStatus();

            if($receiptStatus->isValid()) {
                $latestReceiptInfo = $receiptStatus->getLatestReceiptInfo();
                // You can loop all of them or either get the first one (recently purchased).
                $receiptInfo = $latestReceiptInfo[0];

                $productId = $receiptInfo->getProductId();
                $transactionId = $receiptInfo->getTransactionId();
                $originalTransactionId = $receiptInfo->getOriginalTransactionId();
                $expiresDate = $receiptInfo->getExpiresDate();
                // And so on...
            } else {
                // The receipt is invalid
            }
            
                
        } catch (Exception $e) {
            return response()->json(['success' => false, "message" => "Something went wrong please try again."], 500);
        }

**Comment by imdhemy** [comment]
I'm not sure if I understand the question correctly. 

First of all, The route you declared `/liap/notifications` overrides the route used by the package to handle the server notifications. So, you should change the route to something else.

If you want an example of how to verify a new purchase receipt, check the following:

- [App Store subscription](https://imdhemy.com/laravel-iap-docs/docs/app-store/sell-subscriptions)
- [Google Play subscription](https://imdhemy.com/laravel-iap-docs/docs/google-play/sell-subscriptions)

The following discussion might be helpful as well, #24. Check the selected answer and its replies.

**Reply by alok2911** [reply]
Hello @imdhemy,

Thank you for quick response.

I have created the url in route file can you please checked attached screenshot for the same. Can you please tell me i need a function to get notification code?
![Screenshot_4](https://user-images.githubusercontent.com/119726522/213915104-99ee95f8-6a0b-4e22-8c12-ae9fbcfde4ce.png)

Thanks,

**Reply by imdhemy** [reply]
You can follow the following guides:

1. [Config App store connect](https://imdhemy.com/laravel-iap-docs/docs/server-notifications/app-store).
2. [Create event listeners](https://imdhemy.com/laravel-iap-docs/docs/get-started/event-listeners)

And here is a list of all [available events](https://imdhemy.com/laravel-iap-docs/docs/server-notifications/event-list).
